src/analysis/processing/qgsalgorithmpointstopaths.cpp

```diff
@@ -0,0 +1,358 @@
+                      "If no order expression is specified, the feature ID is used.\n\n"
+                      "A natural sort can be used when sorting by a string attribute "
+                      "or expression (ie. place 'a9' before 'a10').\n\n"
+                      "An attribute or expression can be selected to group points having the same value into the same resulting line." );
+}
+
+QStringList QgsPointsToPathsAlgorithm::tags() const
+{
+  return QObject::tr( "create,lines,points,connect,convert,join" ).split( ',' );
+}
+
+QString QgsPointsToPathsAlgorithm::group() const
+{
+  return QObject::tr( "Vector creation" );
+}
+
+QString QgsPointsToPathsAlgorithm::groupId() const
+{
+  return QStringLiteral( "vectorcreation" );
+}
+
+void QgsPointsToPathsAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterFeatureSource( QStringLiteral( "INPUT" ),
+                QObject::tr( "Input layer" ), QList< int >() << QgsProcessing::TypeVectorPoint ) );
+  addParameter( new QgsProcessingParameterBoolean( QStringLiteral( "CLOSE_PATH" ),
+                QObject::tr( "Create closed paths" ), false, true ) );
+  addParameter( new QgsProcessingParameterExpression( QStringLiteral( "ORDER_EXPRESSION" ),
+                QObject::tr( "Order expression" ), QVariant(), QStringLiteral( "INPUT" ), true ) );
+  addParameter( new QgsProcessingParameterBoolean( QStringLiteral( "NATURAL_SORT" ),
+                QObject::tr( "Sort text containing numbers naturally" ), false, true ) );
+  addParameter( new QgsProcessingParameterExpression( QStringLiteral( "GROUP_EXPRESSION" ),
+                QObject::tr( "Path group expression" ), QVariant(), QStringLiteral( "INPUT" ), true ) );
+  addParameter( new QgsProcessingParameterFeatureSink( QStringLiteral( "OUTPUT" ),
+                QObject::tr( "Paths" ), QgsProcessing::TypeVectorLine ) );
+  // TODO QGIS 4: remove parameter. move logic to separate algorithm if needed.
+  addParameter( new QgsProcessingParameterFolderDestination( QStringLiteral( "OUTPUT_TEXT_DIR" ),
+                QObject::tr( "Directory for text output" ), QVariant(), true, false ) );
+  addOutput( new QgsProcessingOutputNumber( QStringLiteral( "NUM_PATHS" ), QObject::tr( "Number of paths" ) ) );
+
+  // backwards compatibility parameters
+  // TODO QGIS 4: remove compatibility parameters and their logic
+  QgsProcessingParameterField *orderField = new QgsProcessingParameterField( QStringLiteral( "ORDER_FIELD" ),
+      QObject::tr( "Order field" ), QVariant(), QString(), QgsProcessingParameterField::Any, false, true );
+  orderField->setFlags( orderField->flags() | QgsProcessingParameterDefinition::FlagHidden );
+  addParameter( orderField );
+  QgsProcessingParameterField *groupField = new QgsProcessingParameterField( QStringLiteral( "GROUP_FIELD" ),
+      QObject::tr( "Group field" ), QVariant(), QString(), QgsProcessingParameterField::Any, false, true );
```

Comment:
```suggestion
  QgsProcessingParameterField *groupField = new QgsProcessingParameterField( QStringLiteral( "GROUP_FIELD" ),
      QObject::tr( "Group field" ), QVariant(), QStringLiteral( "INPUT" ), QgsProcessingParameterField::Any, false, true );
```

Reply:
Same goes for deprecated `ORDER_FIELD` parameter. But I thought it doesn't matter since it's hidden and used only by scripts.